pygtrie.py

```diff
@@ -237,14 +353,14 @@ def iterate(self, path, shallow, items):
         # Use iterative function with stack on the heap so we don't hit Python's
         # recursion depth limits.
         node = self
-        stack = []
+        stack: List[Iterator[Tuple[str, _Node[V]]]] = []
         while True:
-            if node.value is not _EMPTY:
+            if node.value is not _SENTINEL:
                 yield path, node.value
 
-            if (not shallow or node.value is _EMPTY) and node.children:
+            if (not shallow or node.value is _SENTINEL) and node.children:
                 stack.append(iter(items(node.children)))
-                path.append(None)
+                path.append("")
```

Comment:
Similarly, why is this changed from None to empty string?

Reply:
Hmm, actually empty string isn’t actually correct anyway. What path is can be overridden by subclassess if they decide to overload `_path_from_key`. The only requirement `Trie` class has is that each element of the path (what is called step in the documentation) is hashable.

So `path` passed to the function is a `List[T]`.  Since the method doesn’t know what `T` is, it treats the list as `List[Optional[T]]` and uses `None` value.

I think `_Node` might need to be a generic type over `S` and `V` where `S` is a type which depends on how `_path_from_key` is defined. I’m not really sure what the best approach is.